### PR TITLE
fix(replays): Enforce rate limits

### PR DESCRIPTION
**Pull request by jjbayer**
Rate limits for replays were checked but not enforced. See test case.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@
 - Protocol validation for source map image type. ([#1869](https://github.com/getsentry/relay/pull/1869))
 - Add PHP support for profiling. ([#1871](https://github.com/getsentry/relay/pull/1871))
 
+**Bug Fixes**:
+
+- Enforce rate limits for session replays. ([#1877]https://github.com/getsentry/relay/pull/1877)
```
Comment (olksdr) [inline CHANGELOG.md]:
```suggestion
- Enforce rate limits for session replays. ([#1877](https://github.com/getsentry/relay/pull/1877))
```

File: relay-server/src/utils/rate_limits.rs [hunk 2]
```diff
@@ -562,6 +562,13 @@ where
             return false;
         }
 
+        // Remove replays independently of events
```
Comment (olksdr) [inline relay-server/src/utils/rate_limits.rs]:
```suggestion
        // Remove replays independently of events.
```

File: CHANGELOG.md (hunk 1 above)
Comment (jjbayer) [reply CHANGELOG.md]: Thanks. I should not write markup manually.

**Review by jan-auer — APPROVED** [review]

**Comment by jan-auer** [comment]
Moving ahead with this as the monthly quota is not _counted_ through these rate limits. We need to follow up with correct counting, however.

**Pull request merged**